client/src/pages/Search/index.js

```diff
@@ -11,6 +11,18 @@ class SearchPage extends Component {
     sitspots: [],
   };
 
+  componentDidMount() {
+    const { searchState } = this.props;
+    if (
+      searchState &&
+      !(
+        Object.entries(searchState).length === 0 &&
+        searchState.constructor === Object
+      )
+    )
```

Comment:
hm, the way the `searchState` is done doesn't seem right. It should simply be created, kept and used within the `Search` component, shouldn't be here at all, what is the reason for that? Plus, this is a top level page, so it will always be set to `{}`?

Of course I don't expect you to refactor it at this point, just wanted to point out.

Reply:
Yes the reason is, we want to keep the state of the search between the Home page, and the search page, and I wanted to make the API request only in the search page, so if it has a state on componentDidMount then do the API request

Reply:
ok I think I understand. So essentially you wanna have sort of a "global" state for the component. Redux would be great here, implementing it for yourself is doable but not trivial.. But ok you found a working solution so.. :)